src/fhir/resources.ts

```diff
@@ -169,8 +167,37 @@ export const specimenAndId = (sample: typeof sampleSchema, patientId: string): R
   return { identifier: sample.specimenCode, id: specimen.id, resource: specimen };
 };
 
+export const createNullVariantAndId = (
```

Comment:
As far as I can tell, this function does not get called even when I select the 'set no variant' button in the form. This seems to be because we are not removing the form data corresponding to the variant.

In addition, your `variantExists` variable in `ReportForm.tsx` does not seem to be utilised when submitting the form.

Reply:
Yeah I think we need to add and remove variants in the form, the FormikArray looks like a good call for this. Have set the initial data to not have a variant so that this method is hit by default. Then once we fully implement multiple variants then this will be solved

Reply:
As per my comment above... shall we tackle this in the next PR so that we can make progress?